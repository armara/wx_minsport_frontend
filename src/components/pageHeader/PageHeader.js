import React from 'react'
import PropTypes from 'prop-types'

import 'components/pageHeader/PageHeader.scss'

const PageHeader = ({ title, children }) => (
  <div className="page-header">
    {title === null ? null : <h3>{title}</h3>}
    {children === null ? null : children}
  </div>
)

PageHeader.defaultProps = {
  children: null,
  title: null,
}

PageHeader.propTypes = {
  title: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.func,
  ]),
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
    PropTypes.func,
  ]),
}

export default PageHeader
