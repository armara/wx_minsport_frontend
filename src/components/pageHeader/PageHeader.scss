@import "styles/mixins";

.page-header {
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  align-items: center;
  padding: 2px 20px;
  border-bottom: 1px solid $line-separator;

  h3 {
    @include page-header-text();
    white-space: nowrap;
  }
}