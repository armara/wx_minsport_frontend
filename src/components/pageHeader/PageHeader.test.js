import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import PageHeader from 'components/pageHeader/PageHeader'

configure({ adapter: new Adapter() })

describe('<PageHeader />', () => {
  it('renders correctly default props', () => {
    const tree = renderer
      .create(<PageHeader title="Peter's Great Embassy" />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
