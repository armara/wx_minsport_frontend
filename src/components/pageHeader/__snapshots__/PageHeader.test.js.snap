// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageHeader /> renders correctly default props 1`] = `
<div
  className="page-header"
>
  <h3>
    Peter's Great Embassy
  </h3>
</div>
`;
